xlators/mgmt/glusterd/src/glusterd-geo-rep.c

```diff
@@ -4422,6 +4422,7 @@ glusterd_gsync_read_frm_status(char *path, char *buf, size_t blen)
         return -1;
     }
     ret = sys_read(status_fd, buf, blen - 1);
+    buf[ret] = '\0';
```

Comment:
In case of failure. I think sys_read will return -1 or negative value which will make this line wrong and cause corruption.

Reply:
We should set '\0' in case of success only.

Reply:
Oh yeah. I think the null should be added in the 'if' condition which is which is followed after calling sys_read